yaptide/celery/utils/progress/fluka_monitor.py

```diff
@@ -153,3 +155,28 @@ def read_fluka_out_file(event: threading.Event,
     logging.info("Sending final update for task %d, simulated primaries %d", details.task_id,
                  progress_details.requested_primaries)
     send_task_update(details.simulation_id, details.task_id, details.update_key, up_dict)
+
+
+def read_fluka_file_offline(filepath: Path) -> tuple[int, int]:
+    """Reads fluka out file and returns number of simulated and requested primaries"""
+    simulated_primaries = 0
+    requested_primaries = 0
+    filepath = filepath / "fl_sim001.out"
+    in_progress = False
+    try:
+        with open(filepath, 'r') as f:
```

Comment:
bad idea to modify function argument by overwriting `filepath`
I think better name for function argument would be `output_dir` or something like that

also the name of the function should suggest what you actually read.
Is it the binary (if I remember correctly "unformatted" in Fluka language) estimator output, i.e. `*_fort*` files ?
Or is it general standard output file ? Take a look here 
http://www.fluka.org/fluka.php?id=man_onl&sub=95 or somewhere else in Fluka documentation and use such naming that user would know what that function reads.

Reply:
As for deepsource, maybe it will stop complain if you pass `output_dir / "fl_sim001.out"` to open ?